packages/flutter_test/test/controller_test.dart

```diff
@@ -805,4 +675,278 @@ void main() {
+                label: 'A',
+                child: Semantics(
+                  label: 'B',
+                  child: Semantics(
+                    key: key,
+                    label: 'C',
+                    child: Container(),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        );
+
+        final SemanticsNode node = tester.semantics.get(find.byKey(key));
+        final SemanticsData semantics = node.getSemanticsData();
+        expect(semantics.label, 'A\nB\nC');
+        semanticsHandle.dispose();
+      });
+
+      testWidgets('Does not return partial semantics', (WidgetTester tester) async {
+        final SemanticsHandle semanticsHandle = tester.ensureSemantics();
+        final Key key = UniqueKey();
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: MergeSemantics(
+                child: Semantics(
+                  container: true,
+                  label: 'A',
+                  child: Semantics(
+                    container: true,
+                    key: key,
+                    label: 'B',
+                    child: Container(),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        );
+
+        final SemanticsNode node = tester.semantics.get(find.byKey(key));
+        final SemanticsData semantics = node.getSemanticsData();
+        expect(semantics.label, 'A\nB');
+        semanticsHandle.dispose();
+      });
+    });
+
+    group('simulatedTraversal', () {
+      testWidgets('produces expected traversal', (WidgetTester tester) async {
+        final SemanticsHandle semanticsHandle = tester.ensureSemantics();
+        addTearDown(() => semanticsHandle.dispose());
+
+        await tester.pumpWidget(const MaterialApp(home: _SemanticsTestWidget()));
+
+        expect(tester.semantics.simulatedTraversal(), inOrder(strict: true, <Matcher>[
+          containsSemantics(isHeader: true, label: 'Semantics Test'),
+          containsSemantics(isTextField: true),
+          containsSemantics(label: 'Off Switch'),
+          containsSemantics(hasToggledState: true),
+          containsSemantics(label: 'On Switch'),
+          containsSemantics(hasToggledState: true, isToggled: true),
+          containsSemantics(label: "Multiline\nIt's a\nmultiline label!"),
+          containsSemantics(label: 'Slider'),
+          containsSemantics(isSlider: true, value: '50%'),
+          containsSemantics(label: 'Enabled Button'),
+          containsSemantics(isButton: true, label: 'Tap'),
+          containsSemantics(label: 'Disabled Button'),
+          containsSemantics(isButton: true, label: "Don't Tap"),
+          containsSemantics(label: 'Checked Radio'),
+          containsSemantics(hasCheckedState: true, isChecked: true),
+          containsSemantics(label: 'Unchecked Radio'),
+          containsSemantics(hasCheckedState: true, isChecked: false),
+        ]));
```

Comment:
This is a good example of the syntax for the #107866 use case.

Reply:
Nice!

Can I also do "fuzzy" matching? For example, how would I express that the Node with the label "slider" has to be traversed directly before the node that has `isSlider` set? But I don't want to enforce any other orders in this particular test about other nodes.

Reply:
You can! `containsAllInOrder` will allow fuzzy matching of an iterable (including `Matcher`s) and `orderedEquals` does an exact match. `simulatedTraversal` will also include a `start` and `end` parameter to allow for exact matching of parts of the traversal.